src/Compilers/CSharp/Test/Emit/Emit/BinaryCompatibility.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class BinaryCompatibility : EmitMetadataTestBase
+    {
+        [Fact]
+        public void InvokeVirtualBoundToOriginal()
+        {
+            // A method invocation of a virtual method is statically bound by the C# language to
+            // the original declaration of the virtual method, not the most derived override of
+            // that method. The difference between these two choices is visible in the binary
+            // compatibility behavior of the program at runtime (e.g. when executing the program
+            // against a modified library). This test checks that we bind the invocation to the
+            // virtual method, not the override.
+            var lib0 = @"
+public class Base
+{
+    public virtual void M() { System.Console.WriteLine(""Base0""); }
```

Comment:
The method body (printing "Base0" and "Derived0" could be removed, since never executed by the test.